pyro/distributions/torch/bernoulli.py

```diff
@@ -4,25 +4,24 @@
 
 from pyro.distributions.bernoulli import Bernoulli as _Bernoulli
 from pyro.distributions.torch_wrapper import TorchDistribution
-from pyro.distributions.util import copy_docs_from, get_clamped_probs
+from pyro.distributions.util import copy_docs_from
 
 
 @copy_docs_from(_Bernoulli)
 class Bernoulli(TorchDistribution):
     enumerable = True
 
     def __init__(self, ps=None, logits=None, *args, **kwargs):
-        ps = get_clamped_probs(ps, logits, is_multidimensional=False)
-        torch_dist = torch.distributions.Bernoulli(ps)
-        x_shape = ps.size()
+        torch_dist = torch.distributions.Bernoulli(probs=ps, logits=logits)
+        x_shape = ps.size() if ps is not None else logits.size()
         event_dim = 1
         super(Bernoulli, self).__init__(torch_dist, x_shape, event_dim, *args, **kwargs)
 
     def sample(self):
         return super(Bernoulli, self).sample().type_as(self.torch_dist.probs)
 
     def batch_log_pdf(self, x):
-        return super(Bernoulli, self).batch_log_pdf(x.long())
+        return super(Bernoulli, self).batch_log_pdf(x)
 
     def enumerate_support(self):
```

Comment:
We should fix PyTorch to not convert this to `.long()` ([code](https://github.com/pytorch/pytorch/blob/master/torch/distributions/bernoulli.py#L67)) and then delete this method.

Reply:
Makes sense, will do.